include/invert_quda.h

```diff
@@ -1314,53 +1297,67 @@ namespace quda {
 
   using ColorSpinorFieldSet = ColorSpinorField;
 
-  //forward declaration
   class EigCGArgs;
 
+  //forward declaration
   class IncEigCG : public Solver {
 
   private:
-    Solver *K;
+    const DiracMatrix &matDefl;
+
+    std::shared_ptr<Solver> K;
```

Comment:
The use of smart pointer is appreciated here. But when we introduce them we should make sure to follow https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#S-resource

So a`shared_ptr` should not be the default. I have not checked whether it is necessary here but in general please take a look at the guidelines.